sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuColumnarToRowExec.scala

```diff
@@ -44,10 +45,12 @@ class AcceleratedColumnarToRowIterator(
     numInputBatches: GpuMetric,
     numOutputRows: GpuMetric,
     opTime: GpuMetric,
-    streamTime: GpuMetric) extends Iterator[InternalRow] with Serializable {
+    streamTime: GpuMetric,
+    inTestMode: Boolean = false) extends Iterator[InternalRow] with Serializable {
   @transient private var pendingCvs: Queue[HostColumnVector] = Queue.empty
   // GPU batches read in must be closed by the receiver (us)
   @transient private var currentCv: Option[HostColumnVector] = None
+  @transient private var splitIter: Iterator[Array[ColumnVector]] = Iterator.empty
```

Comment:
Why do we need a `splitIter` in addition to `pendingCvs`?  Wouldn't it be simpler in `setupBatch` to drain the retry iterator and copy them all into `pendingCvs` instead of trying to keep around another iterator to manage?

Reply:
Just want to make the host memory as less as possible.
But I am fine with either of them.

Reply:
updated